leaflet/index.d.ts

```diff
@@ -817,67 +817,73 @@ declare namespace L {
         coordsToLatLng?: (coords: [number, number] | [number, number, number]) => LatLng; // check if LatLng has an altitude property
     }
 
-    /**
-     * Represents a GeoJSON object or an array of GeoJSON objects.
-     * Allows you to parse GeoJSON data and display it on the map. Extends FeatureGroup.
-     */
-    export interface GeoJSON extends FeatureGroup {
-        /**
-         * Adds a GeoJSON object to the layer.
-         */
-        addData(data: GeoJSON.GeoJsonObject): Layer;
-
-        /**
-         * Resets the given vector layer's style to the original GeoJSON style,
-         * useful for resetting style after hover events.
-         */
-        resetStyle(layer: Layer): Layer;
-
-        /**
-         * Changes styles of GeoJSON vector layers with the given style function.
-         */
-        setStyle(style: StyleFunction): this;
-
+    export class GeoJSON {
         /**
          * Creates a Layer from a given GeoJSON feature. Can use a custom pointToLayer
          * and/or coordsToLatLng functions if provided as options.
          */
-        geometryToLayer(featureData: GeoJSON.Feature<GeoJSON.GeometryObject>, options?: GeoJSONOptions): Layer;
+        static geometryToLayer(featureData: GeoJSON.Feature<GeoJSON.GeometryObject>, options?: GeoJSONOptions): Layer;
 
         /**
          * Creates a LatLng object from an array of 2 numbers (longitude, latitude) or
          * 3 numbers (longitude, latitude, altitude) used in GeoJSON for points.
          */
-        coordsToLatLng(coords: [number, number]): LatLng;
-
-        coordsToLatLng(coords: [number, number, number]): LatLng;
+        static coordsToLatLng(coords: [number, number] | [number, number, number]): LatLng;
 
         /**
          * Creates a multidimensional array of LatLngs from a GeoJSON coordinates array.
          * levelsDeep specifies the nesting level (0 is for an array of points, 1 for an array of
          * arrays of points, etc., 0 by default).
          * Can use a custom coordsToLatLng function.
          */
-        coordsToLatLngs(coords: Array<number>, levelsDeep?: number, coordsToLatLng?: (coords: [number, number] | [number, number, number]) => LatLng): LatLng[]; // Not entirely sure how to define arbitrarily nested arrays
+        static coordsToLatLngs(
+            coords: any[],
+            levelsDeep?: number,
+            coordsToLatLng?: (coords: [number, number] | [number, number, number]) => LatLng): any[]; // Using any[] to avoid artificially limiting valid calls
 
         /**
          * Reverse of coordsToLatLng
          */
-        latLngToCoords(latlng: LatLng): [number, number] | [number, number, number];
+        static latLngToCoords(latlng: LatLng): [number, number, number]; // A three tuple can be assigned to a two or three tuple
```

Comment:
This is the best compromise imo. We can't/shouldn't use union types here for the reasons I described in the main comment thread. A three tuple is assignable to a one, two, or three tuple. But a three tuple cannot be assigned to a four (or greater) tuple. Which is more restrictive than just saying it's a number array.